src/swagger/specGenerator2.ts

```diff
@@ -253,6 +253,15 @@ export class SpecGenerator2 extends SpecGenerator {
     return properties;
   }
 
+  protected getSwaggerType(type: Tsoa.Type): Swagger.Schema {
+
+    if (type.dataType === 'union' || type.dataType === 'intersection') {
+      return { type: 'object' };
```

Comment:
I think we should `console.warn` in here some text that says `Swagger 2.0 does not support intersection or unions. If you would like to take advantage of this, please change tsoa.json's "specVersion" to 3.`